src/routes/transactions/entities/swap-order-info.entity.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  TransactionInfo,
+  TransactionInfoType,
+} from '@/routes/transactions/entities/transaction-info.entity';
+import {
+  ApiExtraModels,
+  ApiProperty,
+  ApiPropertyOptional,
+} from '@nestjs/swagger';
+
+export class TokenInfo {
+  @ApiPropertyOptional()
+  logo: string | null;
+
+  @ApiProperty()
+  symbol: string;
+
+  @ApiProperty()
+  amount: string;
+
+  constructor(args: { logo: string | null; symbol: string; amount: string }) {
+    this.logo = args.logo;
+    this.symbol = args.symbol;
+    this.amount = args.amount;
+  }
+}
+
+@ApiExtraModels(TokenInfo)
+export abstract class SwapOrderTransactionInfo extends TransactionInfo {
+  @ApiProperty()
+  status: 'open' | 'fulfilled' | 'cancelled' | 'expired';
```

Comment:
What about exporting these unions into a type? I mean, something like:
```javascript
export type SwapOrderTransactionInfoStatus =
  | 'open'
  | 'fulfilled'
  | 'cancelled'
  | 'expired';

export type DefaultSwapOrderTransactionInfoStatus =
  | 'open'
  | 'cancelled'
  | 'expired';

export type SwapOrderTransactionInfoOrderKind = 'buy' | 'sell';
```

It could be applied to test classes also:
```javascript
.with('status', status as SwapOrderTransactionInfoStatus)
```

Reply:
No strong opinion though, but we could extract the string literals ('open', 'fulfilled', etc.) as well to constants, as they are used in several files (`SwapOrderMapper` for example).

Reply:
For this use-case I actually think that having the union declared in-place is simpler (one less redirection for the reader of what the value might be).

If we declared the union type, we would still need to specify the respective values in `SwapOrderMapper` (maybe with the exception of the `switch/case` where we could use the declared union type).

Reply:
Ok, it makes sense. For `SwapOrderMapper` I was thinking about using not the union but static constants, but I agree with you, maybe it adds more complexity overall. I'm fine with your version, thanks for looking into it!